test/e2e/conftest.py

```diff
@@ -385,6 +387,25 @@ def job_run_as_user() -> PosixSessionUser:
     )
 
 
+@pytest.fixture(scope="session")
+def posix_job_users() -> list:
+    return [
+        PosixSessionUser("job-user", "job-user"),
+        PosixSessionUser(
+            user="cli-override",
+            group="job-override-group",
+        ),
+        PosixSessionUser(
+            user="config-override",
+            group="job-override-group",
+        ),
+        PosixSessionUser(
+            user="env-override",
+            group="job-override-group",
+        ),
+    ]
```

Comment:
I think it makes more sense to split this up. We should be able to select each user without having to memorize/hardcode the `user`/`group` of that user in the test code.

Reply:
What do you mean split this up? Make each user into a different fixture?